Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -309,12 +301,67 @@ bool ShouldRebuild()
                 context);
         }
 
+        // This error is invoked from within the package manager library, which happens in a seperate
+        // process, we scrape stdio for error messages in RestoreAsync method below.
+        private ResolvingErrorReport ReportError = (ErrorReportType errorType, int code, string message) =>
+        {
+            if (errorType == ErrorReportType.Error)
+            {
+                Console.WriteLine("PackageManagement Error {0} {1}", code, message);
+            }
+            else
+            {
+                Console.WriteLine("PackageManagement Warning {0} {1}", code, message);
+            }
+        };
+
+        private object _lockObject = new object();
+        private DependencyProvider _dependencies;
+        private IDependencyManagerProvider _idm;
+        private AssemblyResolutionProbe _assemblyProbingPaths;
+        private NativeResolutionProbe _nativeProbingRoots;
+
+        private void EnsureDependencyProviderLoaded()
+        {
+            if (_dependencies == null)
```

Comment:
Instead of the double-checked lock, can we use `Lazy<DependencyProvider>`?

Reply:
Yes